### PR TITLE
refactor: Refactor the equals function for readability

### PR DESCRIPTION
**Pull request by TehShrike**
- Turn an unnecessary nested if block into an &&
- Change the isRegExp block to use the same "if the second argument is the wrong type, return false" pattern as all the other blocks

### CODE REVIEW

**Comment by gkalpak** [comment]
LGTM (does exactly as the description says - git diffing adds much more noise :smiley:)

**Comment by TehShrike** [comment]
It's true, the [no-whitespace diff](https://github.com/angular/angular.js/pull/13650/files?w=1) is a lot shorter.

File: src/Angular.js [hunk 1]
```diff
@@ -969,38 +969,37 @@ function equals(o1, o2) {
...
-              isDefined(o2[key]) &&
-              !isFunction(o2[key])) return false;
         }
         return true;
       }
+    } else if (isDate(o1)) {
+      if (!isDate(o2)) return false;
+      return equals(o1.getTime(), o2.getTime());
+    } else if (isRegExp(o1)) {
+      if (!isRegExp(o2)) return false;
```
Comment (lgalfaso) [inline src/Angular.js]: [nit] I find `return isRegExp(o2) && o1.toString() == o2.toString();` easier to read, but not a blocker

Comment (TehShrike) [inline src/Angular.js]: Me too, but every other `if (some type)` block used the other pattern, and the function as a whole seemed easier to read when they all followed the same pattern.

**Pull request closed without merging**